File: BackEnd/src/main/java/com/ssafy/backend/db/entity/tactic/Tactic.java
package com.ssafy.backend.db.entity.tactic;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.ssafy.backend.db.entity.user.User;
import com.ssafy.backend.db.entity.game.Game;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name="tactic")
public class Tactic {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    @Column(name="tactic_id")
    private Long tacticId;

    @Column(name="tactic_title", nullable = false)
    private String tacticTitle;

    @Column(name = "tactic_content",nullable = false, length = 10000)
    private String tacticContent;

    @Column(name="local_date_time")
    private LocalDateTime createTime;

    @JsonBackReference
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name="user_id", nullable = false)
    private User user;

    @JsonBackReference
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "game_id", nullable = false)
    private Game game;

}

File: frontend/src/api/Tactic.js
import { api } from "./api";

//tactic
// {
//     "gameId": 1,
//     "tacticContent": "공략글 내용",
//     "tacticTitle": "공략글 제목",
//     "userId": 1
// }

export const postTactics = (tactic) => {
  const url = `tactics`;
  return api.post(url, tactic);
};

export const putTactics = (tactic) => {
  const url = `tactics`;
  return api.put(url, tactic);
};

export const getTacticGame = (gameId) => {
  const url = `tactics/game/${gameId}`;
  return api.get(url);
};

export const getTacticUser = (userId) => {
  const url = `tactics/user/${userId}`;
  return api.get(url);
};
